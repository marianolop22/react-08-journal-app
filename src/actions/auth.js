import { types } from "../types/types";
import { firebase, googleAuthProvider } from "../firebase/firebase-config";
import { finishLoading, startLoading } from "./ui";
import Swal from 'sweetalert2';
import { notesLogoutCleaning } from './notes'

//esto es una accion asincrona
export const startLoginEmailPassword = (email, password) => {
    return ( dispatch ) => {

        dispatch ( startLoading() );

        //se agrega el return para que ser puedan hacer las pruebas
        return firebase.auth().signInWithEmailAndPassword(email, password)
            .then( ({user}) => {
                dispatch (login (user.uid, user.displayName) );
            })
            .catch( err => {
                Swal.fire(
                    'Fail',
                    err.message,
                    'error'
                );
            })
            .finally( ()=> {
                dispatch ( finishLoading()  );
            })
    }
};

export const startRegisterWithEmailPasswordName = (email,  password, name) => {
    
    return ( dispatch ) => {
        return firebase.auth().createUserWithEmailAndPassword(email, password)
            .then( async ({ user }) => {
                await user.updateProfile({displayName: name});
                dispatch ( login ( user.uid, user.displayName ));
            })
            .catch( err => {
                Swal.fire(
                    'Fail',
                    err.message,
                    'error'
                );
            })
    }
};


export const startGoogleLogin = () => {

    return ( dispatch ) => {
        firebase.auth().signInWithPopup( googleAuthProvider )
            .then( ({ user }) => {
                dispatch ( login ( user.uid, user.displayName ));
            })
    }
};



export const startLogout = () => {
    return async ( dispatch ) => {

        // dispatch ( startLoading() );

        await firebase.auth().signOut()
        dispatch ( logout () );
        dispatch ( notesLogoutCleaning ());
        // dispatch ( finishLoading() );
    }
};

export const login = (uid, displayName ) => ({
        type: types.login,
        payload: {
            uid,
            displayName
        }
    })

export const logout = () => ({
    type: types.logout
})