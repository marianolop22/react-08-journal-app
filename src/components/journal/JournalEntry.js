import React from 'react'

export const JournalEntry = () => {
    return (
        <div className="journal__entry">
            <div 
                className="journal__entry-picture"
                style={{
                    backgroundSize:'cover',
                    backgroundImage:'url(https://www.gettyimages.es/gi-resources/images/500px/983801190.jpg)'
                }}    
            >
            </div>

            <div className="journal__entry-body">
                <p className="journal__entry-title">
                    Un nuevo d√≠a
                </p>
                <p className="journal__entry-content">
                    Elit ut ea adipisicing eu eiusmod est officia. Velit culpa elit nulla sint 
                </p>


            </div>

            <div className="journal__entry-date-box">
                <span>Monday</span>
                <h4>28</h4>

            </div>

        </div>
    )
}
