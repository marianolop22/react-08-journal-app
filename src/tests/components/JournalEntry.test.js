import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'; //ES6 modules
import thunk from 'redux-thunk'; 
import { JournalEntry } from '../../components/journal/JournalEntry';
// import { types } from '../../types/types';
import { activeNote } from '../../actions/notes';

//saco este mock y me muestra los parÃ¡metros con los que se llama el active note.
// jest.mock('../../actions/notes', () => ({
//     activeNote: jest.fn(),
// }));

//todo esto es para generar un estado para poder utilizar el redux
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const note = {
    id:'123',
    title: 'un titulo',
    body:'un body',
    date: 0,
    url:'https://hola.min/hola.jpg'
}

const initState = {}


let store = mockStore (initState);
store.dispatch = jest.fn(); //esto es para simular la funcion dispatch

const wrapper = mount(
    <Provider store={store}>
        <JournalEntry { ...note}  />
    </Provider>
);


describe ('Pruebas de Journal Screen', ()=> {

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de dispar el activenote', () => {

        wrapper.find('.journal__entry').prop('onClick')();
        // .simulate('click');
        // expect(activeNote).toHaveBeenLastCalledWith( 
        //     note.id,
        //     {
        //         ...note,
        //     } );

        expect(store.dispatch).toHaveBeenCalledWith(activeNote (
                note.id,
                {
                    ...note,
                }));

    });

});