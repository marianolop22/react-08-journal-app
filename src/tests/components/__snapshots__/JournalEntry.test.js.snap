// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de Journal Screen debe de mostrarse correctamente 1`] = `
<div
  className="journal__entry"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://hola.min/hola.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      un titulo
    </p>
    <p
      className="journal__entry-content"
    >
      un body
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Wednesday
    </span>
    <h4>
      31
    </h4>
  </div>
</div>
`;
